modules/videoio/include/opencv2/videoio/cap_rtx64gige.h

```diff
@@ -0,0 +1,141 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "RTX_GigEV_Camera_Controller.h"
+
+#ifndef CAP_RTX64GIGE
+#define CAP_RTX64GIGE
+#ifdef HAVE_RTX64_GIGE
+
+#define MAX_DRIVER_EVENT_HANDLES            3
+#define GIGEV_EVENT_0_DRIVERSTOPPED         0
+#define GIGEV_EVENT_1_GRAB_STARTED          1
+#define GIGEV_EVENT_2_GRAB_COMPLETE         2
+
+#define CAMERA_PORT                         8888
+#define FRAMERATE                           5
+#define MAX_BYTES_PER_PIXEL                 4
+#define GRAB_TIMEOUT                        1000
+
+#pragma pack(push, 1)
+#pragma pack(show)
+
+
+#pragma pack(pop)
+#pragma pack(show)
+
+
+
+namespace cv
+{
+
+    //
+    //  Gets the offset of the Image Data.  This is for use when sharing Images between RTSS and Windows, using RtGVComm.lib.
+    //
+    unsigned int getImageDataOffset(void);
+
+    //
+    //  Converts a MAC Address to a string.
+    //
+    //  @param pStrMac pointer to the resultant string
+    //  @param pMacAddr pointer to the MAC Address to be converted
+    //
+    void macToString(char * pStrMAC, unsigned char *pMacAddr);
+    
+    //
+    //  Capture class for use with RTX64 GigE
+    //
+    class CvCaptureCAM_Rtx64GigE : public CvCapture
```

Comment:
Okay I will be making changes to use IVideoCapture instead.  However, I have two questions:

1. Since this will likely involve heavily changing all of this source code, would it be better if to use a different pull request, and scrap this one?  Or would you rather I submit these new changes under this pull request once the work is done?

2. We need to save timestamps for images we capture.  Is there some way we could extend the Mat class to also have timestamps?  Is there some other preferable way for us to implement these?

Reply:
`cv::Mat` doesn't support metadata.

You could try to retrieve timestamp via `VideoCapture` property:
- `.grab()` / `.retrieve()`
- `.get(CAP_PROP_TIMESTAMP_NS);` of the current "captured" frame

Main problem here is to properly define timestamp base. May be camera `.open()` timestamp should be zero (or timestamp of the first frame).

Currently there are no examples for this.

---

Feel free to amend and force-push commits - we will squash them into one before merging.